### PR TITLE
ext. msg. email logic into it's own commander + preview endpoint

### PR DESCRIPTION
**Pull request by stkem**

### CODE REVIEW

**Comment by stkem** [comment]
@martinraison (@andrewconner)

File: kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaEmailCommander.scala [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+  case class CleanedMessage(cleanText: String, lookHereTexts: Seq[String], lookHereImageUrls: Seq[String])
+
+  case class ProtoEmail(html: Html, starterName: String, pageTitle: String)
+
+  private def parseMessage(msg: String): CleanedMessage = {
+    val re = """\[((?:\\\]|[^\]])*)\](\(x-kifi-sel:((?:\\\)|[^)])*)\))""".r
+    var textLookHeres = Vector[String]()
+    var imageLookHereUrls = Vector[String]()
+    re.findAllMatchIn(msg).toSeq.toList.foreach { m =>
```
Comment (martinraison) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaEmailCommander.scala]: Is toSeq necessary here?

File: kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaEmailCommander.scala [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+      val segments = m.group(3).split('|')
+      val kind = segments.head
+      val payload = URLDecoder.decode(segments.last, "UTF-8")
+      kind match {
+        case "i" => imageLookHereUrls = imageLookHereUrls :+ payload
+        case "r" => textLookHeres = textLookHeres :+ payload
+        case _ => throw new Exception("Unknow look-here type: " + kind)
```
Comment (martinraison) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaEmailCommander.scala]: Unknow -> Unknown

File: kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaEmailCommander.scala [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+      fromTime.map{ dt =>
+        relevantMessages = relevantMessages.filter{ m =>
+          m.createdAt.isAfter(dt.minusMillis(100))
+        }
+      }
+      toTime.map{ dt =>
+        relevantMessages = relevantMessages.filter{ m =>
+          m.createdAt.isBefore(dt.plusMillis(100))
+        }
+      }
```
Comment (martinraison) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaEmailCommander.scala:132]: My personal preference would go to something like:

```
val relevantMessages = messages.filter { m =>
  (fromTime map { dt => m.createdAt.isAfter(dt.minusMillis(100)) } getOrElse True) &&
  (toTime map { dt => m.createdAt.isBefore(dt.plusMillis(100)) } getOrElse True)
}
```

Feel free to ignore, though

File: kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaEmailCommander.scala [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+      val threadItems = relevantMessages.filterNot(_.from.isSystem).map{ message =>
+      val CleanedMessage(text, lookHereTexts, lookHereImageUrls) = parseMessage(message.messageText)
+      message.from match {
+        case MessageSender.User(id) => ExtendedThreadItem(allUsers(id).shortName, allUsers(id).fullName, Some(allUserImageUrls(id)), text, lookHereTexts, lookHereImageUrls)
+        case MessageSender.NonUser(nup) => {
+          ExtendedThreadItem(nup.shortName, nup.fullName, None, text, lookHereTexts, Seq.empty)
+        }
+        case _ => throw new Exception("Impossible")
+      }
+      }.reverse
```
Comment (martinraison) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaEmailCommander.scala:143]: maybe indent the whole 135-142 block for readability

**Comment by martinraison** [comment]
Nice! lgtm - thanks for working on this

**Pull request merged**